### DISCUSSION TITLE
Confirm the right behavior of user with deleted_at value

### DISCUSSION THREAD

**Discussion by EliMariaaaa** [codeigniter4/shield#1073] [Q&A] [answered]
I created an activate/deactivate functionality for users. I used the soft delete functionality for deactivation. However, when I try to activate the user, it's not shown when called in `$users->findById([user_id])`. And when I remove the `deleted_at` record of the data, I get the record. Is this the correct behavior?

**Comment by kenjis** [comment, marked as answer]
> I created an activate/deactivate functionality for users.

What did you created?

>  I used the soft delete functionality for deactivation.

What do you mean? 

If you use CI4 Model's soft delete feature, it deletes records logically.
If a record has `deleted_at` datetime, the record is treated as deleted.
So `$users->findById([user_id])` returns null.
See https://codeigniter4.github.io/CodeIgniter4/models/model.html#usesoftdeletes
